src/server/server.rs

```diff
@@ -418,22 +418,65 @@ impl<I, E> Builder<I, E> {
     /// }
     /// # }
     /// ```
-    pub fn serve<S, B>(self, new_service: S) -> Server<I, S, E>
+    pub fn serve<M, B>(self, new_service: M) -> Server<I, M, E>
     where
         I: Accept,
         I::Error: Into<Box<dyn StdError + Send + Sync>>,
         I::Conn: AsyncRead + AsyncWrite + Unpin + Send + 'static,
-        S: MakeServiceRef<I::Conn, Body, ResBody = B>,
-        S::Error: Into<Box<dyn StdError + Send + Sync>>,
+        M: MakeServiceRef<I::Conn, Body, ResBody = B>,
+        M::Error: Into<Box<dyn StdError + Send + Sync>>,
         B: HttpBody + 'static,
         B::Error: Into<Box<dyn StdError + Send + Sync>>,
-        E: NewSvcExec<I::Conn, S::Future, S::Service, E, NoopWatcher>,
-        E: ConnStreamExec<<S::Service as HttpService<Body>>::Future, B>,
+        E: NewSvcExec<I::Conn, M::Future, M::Service, E, NoopWatcher>,
+        E: ConnStreamExec<<M::Service as HttpService<Body>>::Future, B>,
     {
         let serve = self.protocol.serve(self.incoming, new_service);
         let spawn_all = serve.spawn_all();
         Server { spawn_all }
     }
+
+    /// Consume this `Builder`, creating a [`Server`](Server) that will run the given service.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # #[cfg(feature = "tcp")]
+    /// # async fn run() {
+    /// use hyper::{Body, Error, Response, Server};
+    /// use hyper::service::{make_service_fn, service_fn};
+    ///
+    /// // Construct our SocketAddr to listen on...
+    /// let addr = ([127, 0, 0, 1], 3000).into();
+    ///
+    /// // Our service used to respond to requests...
+    /// let service = service_fn(|_req| async {
+    ///     Ok::<_, Error>(Response::new(Body::from("Hello World")))
+    /// });
+    ///
+    /// // Then bind and serve...
+    /// let server = Server::bind(&addr).serve_service(service);
+    ///
+    /// // Run forever-ish...
+    /// if let Err(err) = server.await {
+    ///     eprintln!("server error: {}", err);
+    /// }
+    /// # }
+    /// ```
+    pub fn serve_service<S, B>(self, service: S) -> Server<I, Shared<S>, E>
```

Comment:
I don't think the name `serve_service` is particularly good but couldn't come up with anything else.

Reply:
`serve`?

Reply:
What if instead of a separate method, it's part of `hyper::service`?

```rust
Server::bind(addr)
    .serve(hyper::service::shared(svc))
```

Reply:
> `serve`?

`serve` is already taken https://docs.rs/hyper/0.14.7/hyper/server/struct.Builder.html#method.serve 😞 This method is actually not on `Server` but on `server::Builder`. Fixing description now.

> What if instead of a separate method, it's part of `hyper::service`?

I was thinking that since its such a common use case it made sense to have a dedicated method for it.